### PR TITLE
feat(discv5): recycle clean up code

### PR DESCRIPTION
**Pull request by emhane**
Recycles code from https://github.com/paradigmxyz/reth/pull/7681, which clarifies semantics of fork id in the ENR.

### CODE REVIEW

File: crates/net/discv5/src/config.rs [hunk 1]
```diff
@@ -262,6 +267,10 @@ impl BootNode {
     }
 }
 
+/// Auto-trait for value to insert in a local ENR kv-pair.
+pub trait EnrValue: alloy_rlp::Encodable + Debug + Send {}
+impl<T> EnrValue for T where T: alloy_rlp::Encodable + Debug + Send {}
```
Comment (mattsse) [inline crates/net/discv5/src/config.rs:272]: can we store the encoded bytes instead?

don't think this is very useful

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/net/discv5/src/config.rs [hunk 2]
```diff
@@ -31,14 +25,18 @@ pub struct ConfigBuilder {
     discv5_config: Option<discv5::Config>,
     /// Nodes to boot from.
     bootstrap_nodes: HashSet<BootNode>,
-    /// [`ForkId`] to set in local node record.
+    /// Fork kv-pair to set in local node record. Identifies which network/chain/fork the node
+    /// belongs, e.g. (b"opstack", ChainId) or (b"eth", ForkId).
```
Comment (onbjerg) [inline crates/net/discv5/src/config.rs:29]:
```suggestion
    /// belongs, e.g. `(b"opstack", ChainId)` or `(b"eth", ForkId)`.
```

File: crates/net/discv5/src/config.rs [hunk 3]
```diff
@@ -31,14 +25,18 @@ pub struct ConfigBuilder {
...
+    ///
+    /// Defaults to L1 mainnet if not set.
     fork: Option<(&'static [u8], ForkId)>,
     /// RLPx TCP port to advertise. Note: so long as `reth_network` handles [`NodeRecord`]s as
     /// opposed to [`Enr`](enr::Enr)s, TCP is limited to same IP address as UDP, since
     /// [`NodeRecord`] doesn't supply an extra field for and alternative TCP address.
     tcp_port: u16,
-    /// Additional kv-pairs that should be advertised to peers by including in local node record.
-    other_enr_data: Vec<(&'static str, Bytes)>,
+    /// List of (key, rlp-encoded-value) tuples that should be advertised in local node record
```
Comment (onbjerg) [inline crates/net/discv5/src/config.rs:37]:
```suggestion
    /// List of `(key, rlp-encoded-value)` tuples that should be advertised in local node record
```

File: crates/net/discv5/src/config.rs [hunk 4]
```diff
@@ -31,14 +25,18 @@ pub struct ConfigBuilder {
...
+    /// (additional to tcp port, udp port and fork).
```
Comment (onbjerg) [inline crates/net/discv5/src/config.rs:38]:
```suggestion
    /// (in addition to tcp port, udp port and fork).
```

File: crates/net/discv5/src/config.rs [hunk 5]
```diff
@@ -187,12 +187,14 @@ pub struct Config {
     pub(super) discv5_config: discv5::Config,
     /// Nodes to boot from.
     pub(super) bootstrap_nodes: HashSet<BootNode>,
-    /// [`ForkId`] to set in local node record.
+    /// Fork kv-pair to set in local node record. Identifies which network/chain/fork the node
+    /// belongs, e.g. (b"opstack", ChainId) or (b"eth", ForkId).
```
Comment (onbjerg) [inline crates/net/discv5/src/config.rs:191]:
```suggestion
    /// Fork kv-pair to set in local node record. Identifies which network/chain/fork the node
    /// belongs, e.g. `(b"opstack", ChainId)` or `(b"eth", ForkId)`.
```

File: crates/net/discv5/src/config.rs [hunk 6]
```diff
@@ -187,12 +187,14 @@ pub struct Config {
...
     pub(super) fork: (&'static [u8], ForkId),
     /// RLPx TCP port to advertise.
     pub(super) tcp_port: u16,
-    /// Additional kv-pairs to include in local node record.
-    pub(super) other_enr_data: Vec<(&'static str, Bytes)>,
+    /// Additional kv-pairs (asides tcp port, udp port and fork) that should be advertised to
```
Comment (onbjerg) [inline crates/net/discv5/src/config.rs:195]:
```suggestion
    /// Additional kv-pairs (besides tcp port, udp port and fork) that should be advertised to
```

File: crates/net/network/src/config.rs [hunk 7]
```diff
@@ -121,15 +122,19 @@ impl<C> NetworkConfig<C> {
...
 
-        if chain.is_optimism() {
-            builder = builder.fork(OPSTACK, fork_id)
+        if chain.named() == Some(NamedChain::Mainnet) {
+            builder = builder.fork(network_key::ETH, fork_id)
         }
+        // todo: set op EL fork id
+        /*if chain.is_optimism() {
+            builder = builder.fork(network_key::, fork_id)
+        }*/
```
Comment (onbjerg) [inline crates/net/network/src/config.rs:137]: i assume this needs to be handled before merge?

**Review by onbjerg — APPROVED** [review]
smol nits

File: crates/net/network/src/config.rs (hunk 7 above)
Comment (emhane) [reply crates/net/network/src/config.rs:137]: nope, it's left blank for now. OP haven't decided on a network key yet. `'opstack'` is reserved for cl. so no fork id on op el enrs for now.

**Review by mattsse — APPROVED** [review]

**Pull request merged**